docs/linting/TYPED_LINTING.md

```diff
@@ -12,8 +12,7 @@ module.exports = {
   parser: '@typescript-eslint/parser',
   // Added lines start
   parserOptions: {
-    tsconfigRootDir: __dirname,
```

Comment:
I've never seen `tsconfigRootDir` be useful in userland. I know there are cases where there are, but it feels like something we don't need to expend space for in the first-run / getting started docs?

Reply:
Heh, a timely Twitter thread: https://twitter.com/JoshuaKGoldberg/status/1597272609275908097 @shawnmcknight

Reply:
@JoshuaKGoldberg Thanks for tagging me in here and replying to the twitter thread.  This clarified things for me and I was able to get things working.

So in our environment where there's a workspace package for the eslint configuration, I was able to get things working with `parserOptions: { project: './tsconfig.eslint.json' },` and not specifying the `tsconfigRootDir`.

Out of the box, that worked perfectly with my turborepo lint runs which set the `cwd` to the workspace's directory.  I ran into a few issues with the vscode eslint extension and lint-staged so I thought I'd share here in the event it helps with someone else or making docs.  For vscode eslint extension, I found that I _must_ be using vscode workspaces (and have the workspace open) which causes the extension to set the `cwd` to the workspace's directory.  Additionally for `lint-staged`, I had to set up a `lint-staged` configuration file in each workspace, rather than the monorepo root, which enabled `lint-staged` to change the `cwd` to the workspace's directory for the files within that workspace. (Note: This was the proper way to enable `lint-staged` in a monorepo anyway, so it was just cleaning up an improper configuration).  After all this all tooling works -- turborepo lint runs, vscode eslint extension, and lint-staged.

Reply:
Fantastic, thanks for the explanation @shawnmcknight! Filed a docs issue here: #6124